Microsoft.Toolkit.Uwp.UI/Cache/InMemoryStorage.cs

```diff
@@ -136,50 +121,41 @@ public void SetItem(InMemoryStorageItem<T> item)
         /// <returns>Valid item if not out of date or return null if out of date or item does not exist</returns>
         public InMemoryStorageItem<T> GetItem(string id, TimeSpan duration)
         {
-            lock (this)
-            {
-                object key = (object)id;
-                object entry = _inMemoryStorage[key];
-
-                if (entry == null)
-                {
-                    return null;
-                }
+            InMemoryStorageItem<T> tempItem = null;
 
-                var item = (InMemoryStorageItem<T>)entry;
+            if (!_inMemoryStorage.TryGetValue(id, out tempItem))
+            {
+                return null;
+            }
 
-                DateTime expirationDate = DateTime.Now.Subtract(duration);
+            DateTime expirationDate = DateTime.Now.Subtract(duration);
 
-                if (item.LastUpdated > expirationDate)
-                {
-                    return item;
-                }
+            if (tempItem.LastUpdated > expirationDate)
+            {
+                return tempItem;
+            }
 
-                _inMemoryStorage.Remove(key);
+            _inMemoryStorage.TryRemove(id, out tempItem);
```

Comment:
I was mentioning the potential fail of TryRemove due to async op but after reading the link you sent, I'm ok this is fine!